utils/databake/src/primitives.rs

```diff
@@ -35,6 +35,17 @@ where
     T: Bake,
 {
     fn bake(&self, ctx: &CrateEnv) -> TokenStream {
+        if core::mem::size_of::<Self>() == core::mem::size_of::<&[u8]>()
+            && core::any::type_name::<Self>() == core::any::type_name::<&[u8]>()
+        {
+            // Safety: `&T` and `&[u8]` have the same size. Note that `self: &&T`, so this copies
+            // `&T`, not `T` itself.
+            // There are no types smaller than `u8`, so there won't be an alignment or allocation-length
+            // issue even if T is not actually `[u8]`.
+            let bytes: &[u8] = unsafe { core::mem::transmute_copy(self) };
+            let byte_string = syn::LitByteStr::new(bytes, proc_macro2::Span::call_site());
+            return quote!(#byte_string);
```

Comment:
Ideally the clippy allow can live here

Reply:
Ideally it should live on the `&[u8]` expression, but I don't think we can apply it to expressions. Ideally it should not lint.